src/lepton/jpgcoder.cc

```diff
@@ -1533,6 +1533,53 @@ void prep_for_new_file() {
 
 void concatenate_files(int fdint, int fdout);
 
+// ZBR: memory-caching child of ibytestream                      
+struct caching_ibytestream : public ibytestream 
+{
+    static std::vector<unsigned char> g_input_data;  // buffer holding entire contents of the current input file. We made it global in order to avoid memory reallocation for each next file
+    size_t starting_pos;  // initial position in the underlying input stream
+    size_t input_pos;  // current reading position in the input data
+    
+    caching_ibytestream (ibytestream &base_stream) : ibytestream(base_stream)
+    {
+        starting_pos = ibytestream::getsize();
```

Comment:
Why is `starting_pos` set to `getsize()`?

Reply:
caching_ibytestream::getsize() should return current read position in the entire file. This includes bytes read via caching_ibytestream::read*() so far plus bytes read from base_stream prior to creating caching_ibytestream on top of it.

It should be 2, since lepton reads first two bytes initially in order to determine the filetype (JPG/LEP/other) and choose appropriate processing path.

Pushed this entire comment into code to avoid future confusion on this subtle behavior.